crepecake/src/main/java/com/google/cloud/tools/crepecake/builder/BuildAndCacheApplicationLayersStep.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.builder;
+
+import com.google.cloud.tools.crepecake.cache.Cache;
+import com.google.cloud.tools.crepecake.cache.CacheWriter;
+import com.google.cloud.tools.crepecake.cache.CachedLayer;
+import com.google.cloud.tools.crepecake.cache.CachedLayerType;
+import com.google.cloud.tools.crepecake.image.DuplicateLayerException;
+import com.google.cloud.tools.crepecake.image.ImageLayers;
+import com.google.cloud.tools.crepecake.image.LayerBuilder;
+import com.google.cloud.tools.crepecake.image.LayerPropertyNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+// TODO: Add unit test.
```

Comment:
do this now

Reply:
Forgot to remove this TODO. Unit test exists.